### PR TITLE
Add unique constraint for AssetLocation name and facility

### PR DESCRIPTION
**Pull request by Ashesh3**
Required for https://github.com/coronasafe/care_fe/issues/5152

This PR adds a validation check to the `AssetLocation` serializer to ensure that an asset location with the same name and facility does not already exist. The function `validate` in the `AssetLocationSerializer` compares the passed name with the existing entries in the database and raises a `ValidationError` if a match is found.  In addition, the validation now applies to both create and update operations.

The `unique_together` attribute is added to the `AssetLocation` model in the `models.py` file.

The `get_serializer_context` method is also updated to include the facility information when serializing the data.

@coronasafe/code-reviewers

### CODE REVIEW

File: care/facility/api/serializers/asset.py
Comment (vigneshhari) [inline care/facility/api/serializers/asset.py:28]: This is applied across all asset locations, not just in a given facility.

File: care/facility/migrations/0337_auto_20230327_1319.py
Comment (vigneshhari) [inline care/facility/migrations/0337_auto_20230327_1319.py:15]: We cant merge this without checking if there are any current cases like this in production, in which case the migration would fail, @mathew-alex this needs to be checked manually to ensure we don't break anything

**Review by vigneshhari — CHANGES_REQUESTED** [review]
Changes Requested

File: care/facility/migrations/0337_auto_20230327_1319.py
Comment (gigincg) [reply care/facility/migrations/0337_auto_20230327_1319.py:15]: @vigneshhari Wouldn't a Validation do the trick instead of having a Constraint?

Comment (vigneshhari) [reply care/facility/migrations/0337_auto_20230327_1319.py:15]: Just validation should do the trick.

Comment (Ashesh3) [reply care/facility/migrations/0337_auto_20230327_1319.py]: Switched to just validation.

File: care/facility/api/serializers/asset.py
Comment (Ashesh3) [reply care/facility/api/serializers/asset.py]: Removed.

**Comment by Ashesh3** [comment]
Requested changes are made. Only validation is done to check for duplicate locations within the same facility. No constraints.

**Comment by vigneshhari** [comment]
Changes look good, Just confirm if the filter is case sensitive, We want it case insensitive.

**Pull request merged**
